move/sources/utilities.move

```diff
@@ -0,0 +1,314 @@
+module plonk_verifier_addr::utilities_core {
+    use aptos_std::crypto_algebra::{Element, one, mul, deserialize, serialize, scalar_mul, add};
+    use std::bn254_algebra::{Fr, FormatFrMsb, G1, FormatG1Uncompr, G2, FormatG2Uncompr};
+    use std::vector::{length, push_back, append, trim, reverse, slice};
+    use std::vector;
+
+    #[test_only]
+    use aptos_std::crypto_algebra::{eq};
+    #[test_only]
+    use std::hash::sha2_256;
+
+    const ERROR_U256_TO_FR: u64 = 2001;
+    const ERROR_LENGTH_VK: u64 = 2002;
+    const ERROR_LENGTH_PROOF: u64 = 2003;
+    const ERROR_SPHINX_PROOF_VERSION: u64 = 2004;
+
+    const VK_BYTES_SIZE: u64 = 32;
+    const PROOF_CHUNK_BYTE_SIZE: u64 = 32;
+
+    public fun validate_fixture_data(sphinx_proof: vector<u8>, sphinx_vkey: vector<u8>): (vector<u256>, u256) {
+        assert!(length(&sphinx_vkey) == VK_BYTES_SIZE, ERROR_LENGTH_VK);
+        let proof_length = length(&sphinx_proof);
+        assert!(proof_length > 4, ERROR_LENGTH_PROOF);
+        assert!((proof_length - 4) % PROOF_CHUNK_BYTE_SIZE == 0, ERROR_LENGTH_PROOF);
+
+        // convert vkey
+        let vkey: u256 = bytes_to_uint256(sphinx_vkey);
+
+        // check hardcoded plonk verifier hash
+        let expected_version: u256 = 0x7f8918df; // corresponds to v1.0.8.2-testnet artifacts
```

Comment:
@storojs72 Any chance we can put the right-hand side of this in a top-level constant, and simplify the maintenance of this without having to look for magic values in the code?